modules/oe_showcase_list_pages/tests/src/ExistingSite/ListPagesTest.php

```diff
@@ -81,42 +111,100 @@ public function testCreateListPages() {
+    $this->assertSearchResultsTitle('News', 6);
     $this->assertSearchResults([
       'News number 4',
       'News number 5',
       'News number 6',
       'News number 7',
       'News number 8',
       'News number 9',
-      'News number 10',
-      'News number 11',
     ]);
 
-    // Filter results by title.
+    // Filter the News results by title.
     $title_input->setValue('News number 8');
     $search_button->click();
-    $this->assertSearchResultsTitle(1);
+    $this->assertSearchResultsTitle('News', 1);
     $this->assertSearchResults([
       'News number 8',
     ]);
+
+    // Assert only News nodes are part of the result.
+    $title_input->setValue('Event example');
+    $search_button->click();
+    $this->assertSearchResultsTitle('News', 0);
+
+    // Create an Event listing page.
+    $this->drupalGet('node/add/oe_list_page');
+    $page->fillField('Title', 'Event list page');
+    $page->fillField('Source entity type', 'node');
+    $page->fillField('Source bundle', 'oe_sc_event');
+    $page->pressButton('Save');
+
+    $node = $this->getNodeByTitle('Event list page');
+    $this->drupalGet('node/' . $node->id() . '/edit');
+    $page->checkField('Override default exposed filters');
+    $page->checkField('emr_plugins_oe_list_page[wrapper][exposed_filters][oelp_oe_sc_event__title]');
+    $page->checkField('emr_plugins_oe_list_page[wrapper][exposed_filters][oelp_oe_sc_event__oe_sc_event_dates]');
+    $page->pressButton('Save');
+
+    $this->drupalGet('node/' . $node->id());
+
+    // Assert that the unique Event item is displayed.
+    $this->assertSearchResults([
+      'Event example',
+    ]);
+
+    // Assert that the filter form for Events exists.
+    $filter_form = $assert_session->elementExists('css', '#oe-list-pages-facets-form');
+    $title_input = $filter_form->findField('Title');
+    $event_date_input = $filter_form->findField('Event dates');
+    $search_button = $filter_form->find('css', '#edit-submit');
+    $this->assertNotNull($search_button);
+    $this->assertNotNull($title_input);
+    $this->assertNotNull($event_date_input);
+
+    // Filter results by date.
+    $event_date_input->setValue('gt');
+    $date_input = $filter_form->findField('Date');
+    $date_input->setValue('2022-04-03');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 1);
+    $this->assertSearchResults([
+      'Event example',
+    ]);
+
+    // Filter Event results by title.
+    $title_input->setValue('Event');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 1);
+    $this->assertSearchResults([
+      'Event example',
+    ]);
+
+    // Assert only Event nodes are part of the result.
+    $title_input->setValue('News');
+    $search_button->click();
+    $this->assertSearchResultsTitle('Event', 0);
   }
 
   /**
    * Asserts the title above the search results.
    *
+   * @param string $listing_page_content
+   *   Expected List page content type title.
```

Comment:
First, the parameter name `$listing_page_content` is weird.
Second, why not pass the complete node title of the list page?

```suggestion
   * @param string $list_page_title
   *   Title of the list page node.
```